src/runtime_src/hip/core/event.cpp

```diff
@@ -172,13 +164,15 @@ copy_buffer::copy_buffer(std::shared_ptr<stream> s)
 
 bool copy_buffer::submit()
 {
-  //handle = std::async(&xrt::bo::sync, &cbo, cdirection);
+  handle = std::async(static_cast<void (xrt::bo::*)(xclBOSyncDirection)>(&xrt::bo::sync), &cbo, cdirection);
```

Comment:
I think this should work and is clearer:
`std::async([&cbo, cdirection] { cbo.sync(cdirection); });`

Reply:
Hi Soren, Thanks for the comments.
Since cbo and cdirection are member variables of the copy_buffer class, I think I need to capture "this" for the Lambda expression and then use cbo and cdirection through this, otherwise I've been getting compiler error. Please let me know if it is not a good replace.

Reply:
yes, you need to capture "this".  I changed the comment accordingly.

Reply:
Thanks